### PR TITLE
KAFKA-5816: [FOLLOW UP] - create ProducedInternal class

### PR DESCRIPTION
**Pull request by dguy**
Create `ProducedInternal` and remove getters from `Produced`

### CODE REVIEW

**Comment by dguy** [comment]
@guozhangwang, @mjsax, @bbejeck

File: streams/src/main/java/org/apache/kafka/streams/kstream/Produced.java [hunk 1]
```diff
@@ -42,6 +42,12 @@ private Produced(final Serde<K> keySerde,
         this.partitioner = partitioner;
     }
 
+    public Produced(final Produced<K, V> produced) {
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/Produced.java]: Should this be protected? Compare the discussion in https://github.com/apache/kafka/pull/3807#discussion_r137603218

**Review by mjsax — COMMENTED** [review]
Besides the `protected` question LGMT.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java [hunk 2]
```diff
@@ -459,6 +464,11 @@ public void to(final Serde<K> keySerde,
     public void to(final String topic, final Produced<K, V> produced) {
         Objects.requireNonNull(topic, "topic can't be null");
         Objects.requireNonNull(produced, "Produced can't be null");
+        to(topic, new ProducedInternal<>(produced));
+
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java:467]: nit: extra line.

**Comment by bbejeck** [comment]
One question, maybe I missed something from the earlier discussions, but since `ProducedInternal` is only used in `org.apache.kafka.streams.kstream.internals,` shouldn't the class and the access methods be package private?  Otherwise LGTM.

**Comment by guozhangwang** [comment]
LGTM. After addressing the questions from @mjsax and @bbejeck please feel free to merge.

**Comment by dguy** [comment]
merged to trunk

**Pull request closed without merging**